src/vs/workbench/contrib/search/common/searchModel.ts

```diff
@@ -620,30 +625,12 @@ export class FolderMatch extends Disposable {
 		return (this.fileCount() + this.folderCount()) === 0;
 	}
 
-	hasFileUriDownstream(uri: URI): FileMatch | null {
-		const directChildFileMatch = this._fileMatches.get(uri);
-		if (directChildFileMatch) {
-			return directChildFileMatch;
-		}
-
-		const folderMatches = this.folderMatchesIterator();
-		for (const elem of folderMatches) {
-			const match = elem.hasFileUriDownstream(uri);
-			if (match) {
-				return match;
-			}
-		}
-		return null;
+	hasFileUriDownstream(uri: URI): FileMatch | undefined {
+		return this._recursiveFileMatches.get(uri);
 	}
 
 	downstreamFileMatches(): FileMatch[] {
-		let recursiveChildren: FileMatch[] = [];
-		const iterator = this.folderMatchesIterator();
-		for (const elem of iterator) {
-			recursiveChildren = recursiveChildren.concat(elem.downstreamFileMatches());
-		}
-
-		return [...this.fileMatchesIterator(), ...recursiveChildren];
+		return [...this._recursiveFileMatches.values()];
```

Comment:
You could possibly just return an Iterator in this case, so you don't have to create an array